gunagala/imager.py

```diff
@@ -283,6 +284,26 @@ def __init__(self, optic, camera, filters, psf, sky, num_imagers=1, num_per_comp
                 # Not a callable, should be a Simple sky model which just returns AB magnitudes per square arcsecond
                 self.sky_rate[filter_name] = self.SB_to_rate(sb, filter_name)
 
+
+    def set_WCS_centre(self, centre):
+        """
+        Set the WCS CRVALs of the Imager instance to centre.
+
+        Parameters
+        ----------
+        centre : astropy.coordinates.SkyCoord or str
+            Sky coordinates of the image centre. Must be either a SkyCoord object or convertible
+            to one by the constructor of SkyCoord.
+        """
+
+        # Ensure centre is a SkyCoord (this allows entering centre as a string)
+        if not isinstance(centre, SkyCoord):
+            centre = SkyCoord(centre, unit='deg')
```

Comment:
Do you need to specify the unit? The `SkyCoord` constructor can usually work it out for most sensible string formats, and by explicitly specifying it here you're reducing the functionality of the method by requiring that the user always passes `centre` in degree units.  If you do want to be able to explicitly specify units here then it would be better to make that an optional argument of the method so that the user has the option to choose which units to use, or to not specify and let `SkyCoord` work it out.